apps/condo/domains/resident/tasks/notifyResidentsOnPayday.js

```diff
@@ -0,0 +1,123 @@
+const dayjs = require('dayjs')
+const locale_ru = require('dayjs/locale/ru')
+const { get } = require('lodash')
+
+const conf = require('@open-condo/config')
+const { getLogger } = require('@open-condo/keystone/logging')
+const { getSchemaCtx } = require('@open-condo/keystone/schema')
+
+const { BillingReceipt, getPaymentsSum } = require('@condo/domains/billing/utils/serverSchema')
+const { COUNTRIES } = require('@condo/domains/common/constants/countries')
+const { RU_LOCALE } = require('@condo/domains/common/constants/locale')
+const { loadListByChunks } = require('@condo/domains/common/utils/serverSchema')
+const { NEED_TO_PAY_RECEIPT_ON_PAYDAY_MESSAGE_TYPE, BILLING_RECEIPT_ADDED_TYPE, BILLING_RECEIPT_ADDED_WITH_DEBT_TYPE, BILLING_RECEIPT_ADDED_WITH_NO_DEBT_TYPE, BILLING_RECEIPT_AVAILABLE_NO_ACCOUNT_TYPE, BILLING_RECEIPT_AVAILABLE_TYPE, BILLING_RECEIPT_CATEGORY_AVAILABLE_TYPE } = require('@condo/domains/notification/constants/constants')
+const { sendMessage, Message } = require('@condo/domains/notification/utils/serverSchema')
+const { ServiceConsumer } = require('@condo/domains/resident/utils/serverSchema')
+
+
+
+const logger = getLogger('notifyResidentsOnPayday')
+
+const DV_SENDER = { dv: 1, sender: { dv: 1, fingerprint: 'notifyResidentsOnPayday' } }
+const conflictingMessageTypes = [BILLING_RECEIPT_ADDED_TYPE, BILLING_RECEIPT_ADDED_WITH_DEBT_TYPE, BILLING_RECEIPT_ADDED_WITH_NO_DEBT_TYPE, BILLING_RECEIPT_AVAILABLE_NO_ACCOUNT_TYPE, BILLING_RECEIPT_AVAILABLE_TYPE, BILLING_RECEIPT_CATEGORY_AVAILABLE_TYPE, NEED_TO_PAY_RECEIPT_ON_PAYDAY_MESSAGE_TYPE]
+
+async function hasConflictingPushes (context, userId) {
+    const now = dayjs()
+    const userMessages = await Message.getAll(context, {
+        user: { id: userId },
+        createdAt_gte: now.set('hour', 0).set('minute', 0).set('second', 0).toISOString(),
+        createdAt_lte: now.toISOString(),
+    })
+
+    for (const message of userMessages) {
+        if (conflictingMessageTypes.find(type => type === message.type)) return true
+    }
+    return false
+}
+
+async function sendNotification (context, receipt) {
```

Comment:
Is "serverUtils" folder a good place to store this function?

Reply:
I don't think there is much difference. We have several examples of both approaches. On the one hand, you can store all the necessary utilities near the place of their use, if they are not reused anywhere else. You can also put them in separate folders for utilities